source/ota.c

```diff
@@ -2586,6 +2586,11 @@ static IngestResult_t decodeAndStoreDataBlock( OtaFileContext_t * pFileContext,
     /* Decode the file block if space is allocated. */
     if( payloadSize > 0u )
     {
+        if( messageSize <= OTA_FILE_BLOCK_SIZE )
+        {
+            assert( ( payloadSize >= messageSize ) );
+        }
```

Comment:
Yes. your understanding of the problem is correct. If the messageSize is greater than the OTA_FILE_BLOCK_SIZE; we have a different error handling for this. This can be removed and still check for the problem.